### DISCUSSION TITLE
overwrite or delete cached mutation

### DISCUSSION THREAD

**Discussion by joaovitorzv** [TanStack/query#7155] [Q&A] [answered]
I've been wondering how could I handle the following scenario:

say you have a mutation that can update an "order status" and you update the order status 2 times:
 - from "WAITING" to "PACKAGED"
 - then from "PACKAGED" to "READY"
 
this two mutations **if fired while offline** will be cached and persisted on storage, when back online we can resume them and our server will receive two requests updating the order status, but in this case the only request that matters and represents the true status of the order is the last "READY" status.

How could I approach this scenario and keep only one cached mutation, so when resumed it makes a single request directly to the "correct state"? 

except from directly manipulating the `REACT_QUERY_OFFLINE_CACHE` from Async Storage (correct if I'm wrong but I don't think this is a good idea) I couldn't find any way to manipulate cached mutations so I could just check if there is any previous mutation cached and update the variables or even completely remove it and leave only the new one

Am I not supposed to handle this kind of state on the react-query layer? or is there a different way to handle this with react-query?

**Comment by TkDodo** [comment, marked as answer]
I think the scoped mutations proposal could solve this some day, but not immediately:
- https://github.com/TanStack/query/discussions/7126

**Reply by joaovitorzv** [reply]
oh great, the take latest behaviour could solve this... (also thanks for the fast response, love your work!)
